### PR TITLE
zynq: add 5.15 testing kernel

### PR DESCRIPTION
**Pull request by PolynomialDivision**
**Only compile tested.**

### CODE REVIEW

**Comment by luaraneda** [comment]
Thanks a lot for doing this, it was not my TODO list.

Question, did you also update/refresh kernel config?
I usually run the following command after copying the kernel config to make sure kernel config is updated:
`# make kernel_oldconfig`

I expect to have access to a device by mid November.

**Comment by PolynomialDivision** [comment]
> I expect to have access to a device by mid November.

Could you already test it?

**Comment by luaraneda** [comment]
I have access to the device now, so I'll probably test over the weekend.
Can you rebase on top of master? It's currently around 500 commits behind

File: target/linux/zynq/config-5.15 [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+CONFIG_ARM_CPU_SUSPEND=y
+CONFIG_ARM_ERRATA_643719=y
+CONFIG_ARM_ERRATA_720789=y
+CONFIG_ARM_ERRATA_754322=y
+CONFIG_ARM_ERRATA_754327=y
+CONFIG_ARM_ERRATA_764369=y
+CONFIG_ARM_ERRATA_775420=y
+CONFIG_ARM_GIC=y
+CONFIG_ARM_GLOBAL_TIMER=y
+CONFIG_ARM_GT_INITIAL_PRESCALER_VAL=1
```
Comment (luaraneda) [inline target/linux/zynq/config-5.15:32]: From looking at the kernel sources, it seems the apropiated value is 2.
drivers/clocksource/Kconfig:
```
config ARM_GT_INITIAL_PRESCALER_VAL
	int "ARM global timer initial prescaler value"
	default 2 if ARCH_ZYNQ
	default 1
	depends on ARM_GLOBAL_TIMER
```
Default of 2 comes from kernel 5.14 commit 68e2215e9d5f5ec8e5ba0158683742932519cad9:
(arm: zynq: don't disable CONFIG_ARM_GLOBAL_TIMER due to CONFIG_CPU_FREQ anymore)

File: target/linux/zynq/config-5.15 [hunk 2]
```diff
@@ -0,0 +1,553 @@
...
+CONFIG_I2C_MUX=y
+CONFIG_I2C_MUX_PCA954x=y
+CONFIG_ICST=y
+CONFIG_IIO=y
+CONFIG_IIO_BUFFER=y
+CONFIG_IIO_KFIFO_BUF=y
+CONFIG_IIO_TRIGGER=y
+CONFIG_IIO_TRIGGERED_BUFFER=y
+CONFIG_IKCONFIG=y
+CONFIG_IKCONFIG_PROC=y
```
Comment (luaraneda) [inline target/linux/zynq/config-5.15:251]: They are set as "is not set" on generic config, were set on k5.10 config, but they were removed from target-specific configs on 9995dd254374aa627257ab3218512877a9ae33de
They are also provied by the ikconfig kernel package:
package/kernel/linux/modules/other.mk:
```
define KernelPackage/ikconfig
  SUBMENU:=$(OTHER_MENU)
  TITLE:=Kernel configuration via /proc/config.gz
  KCONFIG:=CONFIG_IKCONFIG \
           CONFIG_IKCONFIG_PROC=y
  FILES:=$(LINUX_DIR)/kernel/configs.ko
  AUTOLOAD:=$(call AutoLoad,70,configs)
endef
```

**Review by luaraneda — APPROVED** [review]
Changes from this PR are working fine on the device I tested (Digilent Zybo Z7-20).

I do have comments about some config symbols, please review them.
They were found by copying k5.10 config and running "make kernel_oldconfig".

Feel free to add the following tags:
```
Reviewed-by: Luis Araneda <luaraneda@gmail.com>
Tested-by: Luis Araneda <luaraneda@gmail.com> # Digilent Zybo Z7-20
```

File: target/linux/zynq/config-5.15 (hunk 1 above)
Comment (PolynomialDivision) [reply target/linux/zynq/config-5.15]: changed that. thanks!

File: target/linux/zynq/config-5.15 (hunk 2 above)
Comment (PolynomialDivision) [reply target/linux/zynq/config-5.15]: changed it. thanks!

**Comment by PolynomialDivision** [comment]
@hauke @Ansuel I think this is ready to be merged.

**Comment by PolynomialDivision** [comment]
Now it's ready to be merged.

**Comment by luaraneda** [comment]
I think that after a short period we should make 5.15 the current kernel version.
That would allow us to catch any issue sooner while having the ability to rollback to 5.10.

**Comment by ynezz** [comment]
Thanks, merged in bb39ed3f36d26

**Pull request closed without merging**